crates/cheatcodes/src/evm.rs

```diff
@@ -600,6 +653,115 @@ pub(super) fn get_nonce<DB: DatabaseExt>(ccx: &mut CheatsCtxt<DB>, address: &Add
     Ok(account.info.nonce.abi_encode())
 }
 
+fn update_gas_snapshot<DB: DatabaseExt>(
+    ccx: &mut CheatsCtxt<DB>,
+    group: Option<&String>,
+    name: String,
+    value: String,
+) -> Result {
+    // Create the snapshot if it doesn't exist
+    create_dir_all(ccx.state.config.paths.snapshots.clone())?;
+
+    if let Some(group) = group {
+        // Prepare the snapshot.
+        // Check if the snapshot already exists, if so, update the value in place.
+        // Otherwise, create a new snapshot.
+        let snapshot_path = &ccx.state.config.paths.snapshots.join(format!("{}.json", group));
+        let mut snapshot: BTreeMap<String, String> =
+            read_json_file(&snapshot_path).unwrap_or_else(|_| BTreeMap::new());
+        snapshot.insert(name.clone(), value);
+
+        // Write the snapshot to a file, asserting that the write was successful.
+        let result = write_pretty_json_file(&snapshot_path, &snapshot).is_ok();
```

Comment:
Love the idea of groups - one thought is, does BTreeMap serialize deterministically? Given test run ordering isn't deterministic afaik, worried thrashing of snapshot output will make diffs more difficult to read

Reply:
I guess since BTreeMap is ordered on `name` and presumably serde_json just iterates over it, it's likely deterministic

Reply:
Good point!

Given that upon applying an update the `json` is read into a BTreeMap, changes are applied and then serialized again the order should be deterministic. I'll need to add more tests to confirm this assumption is correct, but that is the goal.